CMakeLists.txt

```diff
@@ -102,7 +102,7 @@ if(OPENDDS_XERCES3)
 endif()
 if(OPENDDS_SECURITY)
   add_subdirectory(dds/DCPS/security)
-  if(OPENDDS_BUILT_IN_TOPICS)
+  if(OPENDDS_BUILT_IN_TOPICS AND NOT OPENDDS_CXX_STD_YEAR LESS 2011)
     add_subdirectory(tools/dds/rtpsrelaylib)
     add_subdirectory(tools/rtpsrelay)
```

Comment:
I don't mind this strategy but I think we should consider (maybe for a future PR) adding all the subdirectories and letting per-target properties determine what gets built

Reply:
This pattern comes from the CMake tests. I think the rational was if you wanted to configure a project individually, it would check the requirements and fail to configure. You can't really configure the non-test OpenDDS projects individually though, so maybe... The only issue I have with it is I'd like the tests current setup to remain because I think it definitely makes sense for the tests. So having having them act differently might be confusing.

Reply:
I think it would be easier to manage if it were consistently done on a target/properties evaluation basis and not imperative logic around which subdirectories are added.  A variable that's something like `BUILD_THE_TESTS` is very clearly tied to `add_subdirectory(tests)` so that's fine.